src/main/java/io/aiven/kafka/auth/json/AivenAcl.java

```diff
@@ -38,26 +38,34 @@ public class AivenAcl {
     @SerializedName("resource_pattern")
     public final String resourceRePattern;
 
+    @SerializedName("permission_type")
+    private final AclPermissionType permissionType;
+
     public AivenAcl(final String principalType,
                     final String principal,
                     final String operation,
                     final String resource,
-                    final String resourcePattern) {
+                    final String resourcePattern,
+                    final AclPermissionType permissionType) {
         this.principalType = principalType;
         this.principalRe = Pattern.compile(principal);
         this.operationRe = Pattern.compile(operation);
-        this.resourceRe = Objects.nonNull(resource)
-            ? Pattern.compile(resource) : null;
+        this.resourceRe = Objects.nonNull(resource) ? Pattern.compile(resource) : null;
         this.resourceRePattern = resourcePattern;
+        this.permissionType = Objects.isNull(permissionType) ? AclPermissionType.ALLOW : permissionType;
+    }
+
+    public AclPermissionType getPermissionType() {
+        return permissionType == null ? AclPermissionType.ALLOW : permissionType;
     }
```

Comment:
We still need the getter because Gson does not call `AclPermissionTypeDeserializer` if `permission_type` is not set at all in the json, so there's still the change that `permissionType` will be null.

Reply:
Oh, I see. Yeah, sure, let's leave it as is now in this case. Could you please leave a comment somewhere explaining this, otherwise someone will "optimize" :)

Reply:
Added the comment